repo/lammps/package.py

```diff
@@ -9,15 +9,12 @@
 
 class Lammps(BuiltinLammps):
 
-  version("stable_29Aug2024_update3", tag="stable_29Aug2024_update3")
+  variant("pace", default=False, description="Enable the ML PACE module")
+  depends_on("pace", when="+pace")
 
   depends_on("kokkos+openmp cxxstd=17", when="+openmp")
-  depends_on("kokkos+rocm", when="+rocm")
-  depends_on("kokkos+cuda cxxstd=17", when="+cuda")
+  depends_on("kokkos+wrapper", when="+cuda")
   
```

Comment:
@scheibelp @pearce8 How to rewrite this spack clause in so that it satisfies the following condition in the `kokkos/package.py`
```
if spec.satisfies("~wrapper+cuda") and not (spec.satisfies("%clang") or spec.satisfies("%cce") 
or spec.satisfies("+cmake_lang")):  
        raise InstallError("Kokkos requires +wrapper when using +cuda without %clang, %cce or +cmake_lang")   
```

Reply:
I think this indicates missing constraints in `kokkos`, for now you can add something to our extension of kokkos like:

```
class Kokkos(BuiltinKokkos):
    requires("+wrapper", when="+cuda ~cmake_lang %gcc")
    # ... and so on for each compiler of interest other than cce and clang
```

(at which point the spec will automatically concretize to `+wrapper` on Matrix. If we are trying to avoid `+wrapper`, we can add `require: +cmake_lang` to kokkos on Matrix)